src/sentry/api/endpoints/useravatar.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import absolute_import
+
+from PIL import Image
+
+from django.conf import settings
+
+from rest_framework import status
+from rest_framework.response import Response
+
+from sentry.api.bases.user import UserEndpoint
+from sentry.api.serializers import serialize
+from sentry.models import UserAvatar, File
+from sentry.utils.compat import StringIO
+
+
+MIN_DIMENSION = 256
+
+MAX_DIMENSION = 1024
+
+
+class UserAvatarEndpoint(UserEndpoint):
+    FILE_TYPE = 'avatar.file'
+
+    def get(self, request, user):
+        return Response(serialize(user, request.user))
+
+    def is_valid_size(self, width, height):
+        if width != height:
+            return False
+        if width < MIN_DIMENSION:
+            return False
+        if width > MAX_DIMENSION:
+            return False
+        return True
+
+    def put(self, request, user):
+        if user != request.user:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        photo_string = request.DATA.get('avatar_photo')
+        photo = None
+        if photo_string:
+            photo_string = photo_string.decode('base64')
+            if len(photo_string) > settings.SENTRY_MAX_AVATAR_SIZE:
+                return Response(status=status.HTTP_400_BAD_REQUEST)
```

Comment:
This should properly be a 413 response code to indicate that it's too large, rather than the request was just bad. I'm assuming you could use this on the frontend to error gracefully and prompt "image is too large" or something.